### PR TITLE
feat: start Reviewer port

### PR DESCRIPTION
**Pull request by BrayanDSO**
## Purpose / Description
I was going to finish it locally before pushing to avoid churn if I decide to refactor something, but I ended up procrastinating because of it. So, done is better than perfect.

It is currently a developer option. The design features are going to be implemented incrementally to ease the reviewing process and discussion.

When enough features are implemented, my plan is to move it into `Settings > Advanced > Experimental`

Right now it rate cards and exits when finished. Awesome, isn't it?

## Fixes
* Part of #14302 

## Approach

In the commits

## How Has This Been Tested?

Emulator 34:

<details>
<summary>Light theme screenshot</summary>

Screenshot_20240522_144052

</details>

[reviewer.webm](https://github.com/ankidroid/Anki-Android/assets/69634269/bad6b158-0b6a-4d75-9d6f-42e1729be4cd)

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [X] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code
- [X] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by BrayanDSO** [comment]
I haven't chosen the dark mode colors for the answer buttons yet, and choosing colors take time. Open to suggestions, but definitely don't want to bikeshed the PR because of colors. That always can be changed later

**Comment by criticalAY** [comment]
UI: (Ik its in development mode but still pointing it out just in case) The answer button are bit overflowing ig ?

**Comment by BrayanDSO** [comment]
> The answer button are bit overflowing ig ?

idk what you meant

**Comment by criticalAY** [comment]
I meant 
image
 the layout above and the buttons layout is not aligned here

**Comment by criticalAY** [comment]
> Current layout plan: 

This is the updated one in the PR or you plan to have it this way, as it has perfect alignment

**Comment by BrayanDSO** [comment]
didn't copy that part when branching. Updated it. Thanks!

File: AnkiDroid/src/main/java/com/ichi2/anki/ui/windows/reviewer/ReviewerViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+class ReviewerViewModel(cardMediaPlayer: CardMediaPlayer) :
+    CardViewerViewModel(cardMediaPlayer),
+    PostRequestHandler {
+
+    private var queueState: Deferred<CurrentQueueState?> = asyncIO {
+        withCol { sched.currentQueueState() }
+    }
+    override var currentCard = asyncIO {
+        // this assumes that the Reviewer won't be launched if there isn't a queueState
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/ui/windows/reviewer/ReviewerViewModel.kt:47]: Why don't we move the `!!` further up the call stack?

**Review by david-allison — COMMENTED** [review]
LGTM, one question

Note: * https://github.com/ankidroid/Anki-Android/issues/15695 needs to be prioritized

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/ReviewerFragmentTest.kt [hunk 2]
```diff
@@ -0,0 +1,184 @@
+/*
+ *  This program is free software; you can redistribute it and/or modify it under
```
Comment (david-allison) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/ReviewerFragmentTest.kt:2]: No copyright?

**Review by david-allison — APPROVED** [review]
LGTM, with the understanding that this is development only

No point in blocking

Only encouragement is to add more tests - now is a great opportunity to improve coverage

**Comment by snowtimeglass** [comment]
> Current layout plan:

(In case of that standard screen size,) is the maximum number of icons that can be displayed on the app bar supposed to be two?

**Comment by BrayanDSO** [comment]
> Is the maximum number of icons that can be displayed on the app bar supposed to be two?

No. If you have any points about it, discuss it in Discord, as this PR doesn't do any of that.

**Comment by lukstbit** [comment]
Also LGTM. UI design issues can be discussed later.

**Review by lukstbit — APPROVED** [review]

**Pull request merged**